BonusCalcApi.Tests/V1/Controllers/OperativesControllerTests.cs

```diff
@@ -55,6 +59,39 @@ public void SetUp()
             _classUnderTest.ProblemDetailsFactory = _problemDetailsFactoryMock.Object;
         }
 
+        [Test]
+        public async Task GetOperativesReturnsOk()
+        {
+            // Arrange
+            var expectedOperatives = _fixture.CreateMany<Operative>();
+            _getOperativesUseCaseMock
+                .Setup(m => m.ExecuteAsync(It.IsAny<string>(), It.IsAny<int>(), It.IsAny<int>()))
+                .ReturnsAsync(expectedOperatives);
+
+            // Act
+            var objectResult = await _classUnderTest.GetOperatives("12345678", 1, 25);
+            var statusCode = GetStatusCode(objectResult);
+            var result = GetResultData<List<OperativeResponse>>(objectResult);
+
+            // Assert
+            statusCode.Should().Be((int) HttpStatusCode.OK);
+            result.Should().BeEquivalentTo(expectedOperatives.Select(pe => pe.ToResponse()).ToList());
+        }
+
+        [Test]
+        public async Task GetOperativesReturnsBadRequestIfQueryIsMissing()
+        {
+            // Arrange
+
+            // Act
+            var objectResult = await _classUnderTest.GetOperatives("", null, null);
+            var statusCode = GetStatusCode(objectResult);
+
+            // Assert
+            statusCode.Should().Be((int) HttpStatusCode.BadRequest);
+            _problemDetailsFactoryMock.VerifyStatusCode(HttpStatusCode.BadRequest);
+        }
+
```

Comment:
Similar to the existing test GetOperativeReturnsNotFoundIfPayrollNumberNotFound, you could include a test to confirm a 404 response if the query doesn't find anything - especially as it looks like you're doing a full-text search.

Reply:
It returns an empty array if no operatives match the query which typically returns 200 rather than 404 response - we'd need to check for the empty list and then return the 404 ourselves. Happy to do that if that's what other Hackney APIs do.

Reply:
Nah, it's OK as it is, just a suggestion to keep it consistent with the GetById ep.